pennylane/devices/default_qubit.py

```diff
@@ -95,7 +95,7 @@ def stopping_condition(op: qml.operation.Operator) -> bool:
     if op.__class__.__name__[:3] == "Pow" and qml.operation.is_trainable(op):
         return False
 
-    return op.has_matrix
+    return isinstance(op, (Conditional, MidMeasureMP)) or op.has_matrix
```

Comment:
I just noticed that `adjoint_ops` (stopping condition for decomposing when using adjoint on line 173) only verifies that the tape doesn't have `MidMeasureMP`, but not `Conditional`. Could you add that too?

Reply:
Nice catch, will do.